### PR TITLE
Update navigation to new vanilla pattern with new items

### PR DESCRIPTION
**Pull request by petesfrench**
## Done

Update navigation to new vanilla pattern & new items based on the items listed in [the issue](https://github.com/canonical/web-squad/issues/6054) and [the wireframe](https://miro.com/app/board/o9J_luilNcw=/?moveToWidget=3458764526315426035&cot=14)

Based off of this design:
image

## QA

- Go to https://canonical-com-650.demos.haus/
- Check the design matches the [new vanilla pattern](https://vanillaframework.io/docs/patterns/navigation)
- Check all the items listed in [the issue](https://github.com/canonical/web-squad/issues/6054) are present

## Issue / Card

Fixes https://github.com/canonical/web-squad/issues/6054

### CODE REVIEW

**Comment by spencerbygraves** [comment]
@petesfrench I think you could extend the hover of indented items to full width, and keep the text the same style as the level above. Thanks.

**Comment by petesfrench** [comment]
Thanks @spencerbygraves , could you take another look please

**Comment by carkod** [comment]
@petesfrench are you merging into the wrong branch, should be main?

**Comment by petesfrench** [comment]
@carkod It was requested alongside work on the application page that is already on main. @marinacastejon Do you know if this is supposed to go to production?

**Comment by carkod** [comment]
> @carkod It was requested alongside work on the application page that is already on main. @marinacastejon Do you know if this is supposed to go to production?

Sorry, I meant "it shouldn't be main"? Because the about us page is going 404. We probably don't want this to happen in prod, right?

**Comment by spencerbygraves** [comment]
Thanks @petesfrench, I think that looks better.

**Review by sowasred2012 — CHANGES_REQUESTED** [review]
A lot of the broken links exist currently on the site:
Company:
- About: drop this one, there is nothing we can use at the moment
- Blog: /blog
- Documentation:
- Press centre: /press-centre

Partners:
- About: drop for now
- Our partner programmes: /partners
- Channel/reseller: /partners/channel-and-reseller
- IoT: /partners/iot-device
- Log in to the partner portal: https://partner-portal.canonical.com/
- Also, a number of these pages have a white strip above the main suru that can be gotten rid of:
 image

Careers:
- Overview: /careers
- Open Roles: /careers/all
- Career finder: /careers/start (this is 500ing on the demo, not sure why)
- Working at Canonical: the closest we currently have is /careers/departments

File: static/sass/styles.scss [hunk 1]
```diff
@@ -232,3 +230,13 @@ $p-small-lh-diff: map-get($line-heights, default-text) -
     }
   }
 }
+
+// Style drowdown sub-links to match the other links in the navigation
+ul.p-navigation__dropdown-sub-list {
```
Comment (sowasred2012) [inline static/sass/styles.scss:235]: @petesfrench I think it's worth opening an issue for this in Vanilla, I'm surprised it doesn't already have something for this use case (or maybe it does, but it's hidden away and needs to be documented).

File: templates/partial/_navigation.html [hunk 2]
```diff
@@ -1,37 +1,91 @@
...
-          role="menuitem">
-          <a class="p-navigation__link" href="/products">Products</a>
+        <li class="p-navigation__item--dropdown-toggle" id="company">
+          <a href="/company" aria-controls="company-menu" class="p-navigation__link">Company</a>
+          <ul class="p-navigation__dropdown" id="company-menu" aria-hidden="true">
+            <li>
+              <a href="/company/about-us" class="p-navigation__dropdown-item">About us</a>
+            </li>
+            <li>
+              <a href="/company/blogs" class="p-navigation__dropdown-item">Blogs</a>
```
Comment (sowasred2012) [inline templates/partial/_navigation.html:28]:
```suggestion
              <a href="/company/blogs" class="p-navigation__dropdown-item">Blog</a>
```

File: templates/partial/_navigation.html [hunk 3]
```diff
@@ -1,37 +1,91 @@
...
     </div>
-    <nav class="p-navigation__nav">
+
+    <nav class="p-navigation__nav" aria-label="main-navigation">
       <ul class="p-navigation__items">
-        <li
-          class="p-navigation__item {% if request.path|get_nav_path == 'products' %}is-selected{% endif %}"
-          role="menuitem">
-          <a class="p-navigation__link" href="/products">Products</a>
+        <li class="p-navigation__item--dropdown-toggle" id="company">
```
Comment (sowasred2012) [inline templates/partial/_navigation.html:21]: @spencerbygraves @lyubomir-popov @al-bakalova - can we get some advice on this please - if a navigation item is within a dropdown menu, how do we indicate the currently active nav item to the user?

Peek 2022-10-14 11-48

My thought is to add `.is-selected` to the dropdown toggle, and then an active state to the links (which I'm not sure exists in Vanilla). What do you think?

Comment (sowasred2012) [reply templates/partial/_navigation.html:21]: Demo here: https://canonical-com-650.demos.haus/

**Comment by lyubomir-popov** [comment]
@sowasred2012 my proposal: 
image

**Comment by petesfrench** [comment]
@lyubomir-popov I have tried to replicate what you provided as best I could, can you take a look and give me some feedback please

**Review by sowasred2012 — COMMENTED** [review]
Looking good! The only thing remaining is that the "Products" item in the nav doesn't scroll to the Products section for some reason

**Review by sowasred2012 — APPROVED** [review]
Nice! :rocket:

**Pull request merged**